apps/aecore/test/aecore_forking_SUITE.erl

```diff
@@ -0,0 +1,435 @@
+     {group, all_nodes}
+    ].
+
+groups() ->
+    [
+     {all_nodes, [sequence], [{group, two_nodes}]},
+     {two_nodes, [sequence],
+      [create_dev1_chain,
+       create_dev2_chain,
+       sync_fork_in_wrong_order]}
+    ].
+
+suite() ->
+    [].
+
+init_per_suite(Config) ->
+    ok = application:ensure_started(erlexec),
+    DataDir = ?config(data_dir, Config),
+    TopDir = top_dir(DataDir),
+    Config1 = [{top_dir, TopDir}|Config],
+    make_shortcut(Config1),
+    ct:log("Environment = ~p", [[{args, init:get_arguments()},
+                                 {node, node()},
+                                 {cookie, erlang:get_cookie()}]]),
+    create_configs(Config1),
+    make_multi(Config1),
+    Config1.
+
+end_per_suite(Config) ->
+    stop_node(dev1, Config),
+    stop_node(dev2, Config),
+    ok.
+
+init_per_group(two_nodes, Config) ->
+    [{nodes, [n(dev1),
+              n(dev2)]} | Config];
+init_per_group(_Group, Config) ->
+    Config.
+
+end_per_group(_Group, _Config) ->
+    ok.
+
+init_per_testcase(_Case, Config) ->
+    ct:log("testcase pid: ~p", [self()]),
+    [{tc_start, os:timestamp()}|Config].
+
+end_per_testcase(_Case, Config) ->
+    Ts0 = ?config(tc_start, Config),
+    ct:log("Events during TC: ~p", [[{N, all_events_since(N, Ts0)}
+                                     || {_,N} <- ?config(nodes, Config)]]),
+    ok.
+
+%% ============================================================
+%% Test cases
+%% ============================================================
+
+create_dev1_chain(Config) ->
+    start_node_(dev1, Config),
+    N1 = node_(dev1),
+    connect(N1),
+    Blocks = [ mine_one_block(N1) || _ <- lists:seq(1,8)],
+    true = (length(lists:usort(Blocks)) >= 4),
+    N1Top = lists:last(Blocks),
+    ct:log("top of chain dev1: ~p", [ N1Top ]),
+    stop_node(dev1, Config),   %% make sure we do not sync with dev2.
+    ok.
+
+create_dev2_chain(Config) ->
+    start_node_(dev2, Config),
+    N2 = node_(dev2),
+    connect(N2),
+    mine_one_block(N2),
+    {ok, ForkTop} = rpc_call(N2, aec_chain, top, []),
+    ct:log("top of fork dev2: ~p", [ ForkTop ]),
+    stop_node(dev2, Config),
+    ok.
+
+sync_fork_in_wrong_order(Config) ->
+    start_node_(dev1, Config),
+    N1 = node_(dev1),
+    connect(N1),
+    {ok, N1Top} = rpc_call(N1, aec_chain, top, []),
+    ct:log("top of chain dev1: ~p", [ N1Top ]),
+    stop_node(dev1, Config),
+   
+    start_node_(dev2, Config),
+    N2 = node_(dev2),
+    connect(N2),
+    {ok, ForkTop} = rpc_call(N2, aec_chain, top, []),
+    ct:log("top of chain dev2: ~p", [ ForkTop ]),
+    
+    false = (ForkTop == N1Top),
+    timer:sleep(100),
+    %% unexepctedly last block of dev1 arrives before rest of the chain
+    rpc_call(N2, aec_conductor, post_block, [N1Top]), 
+
+    start_node_(dev1, Config),
+    connect(N1),
+    timer:sleep(200),
+
+    {ok, N2Top} = rpc_call(N2, aec_chain, top, []),
+    ct:log("top of chain dev2: ~p", [ N2Top ]),
+    true = (N1Top == N2Top),
```

Comment:
Errors could be made more self-descriptive: https://github.com/lucafavatella/epoch/commit/d2861c126f039c9ee9ca4fc04d085a573a602652
(Not needed for PR, I am using it locally for testing.)